flood_adapt/dbs_controller.py

```diff
@@ -133,13 +154,23 @@ def plot_slr_scenarios(self) -> str:
         ) as e:
             print(e)
 
+        ref_year = self.site.attrs.slr.relative_to_year
+        if ref_year > df["Year"].max() or ref_year < df["Year"].min():
+            raise ValueError(
+                f"The reference year {ref_year} is outside the range of the available SLR scenarios"
+            )
+        else:
+            scenarios = self.get_slr_scn_names()
+            for scn in scenarios:
+                ref_slr = np.interp(ref_year, df["Year"], df[scn])
+                df[scn] -= ref_slr
+
         df = df.drop(columns="units").melt(id_vars=["Year"]).reset_index(drop=True)
         # convert to units used in GUI
-        slr_current_units = UnitfulLength(value=df.iloc[0, -1], units=units)
+        slr_current_units = UnitfulLength(value=1.0, units=units)
```

Comment:
Wouldn't it be prettier if we moved most of the unit conversion to the type class (the conversion factor has nothing to do with the value so calculating the conversion factor can be moved to in this case UnitTypesLength). In that case, you also don't need the silly unit length declaration to get the conversion factor

Reply:
Check next comment first, moving it might still be prettier but that is only for the lower level class

Reply:
I'll make an issue out of it. Kathryn needs to have this fixed for the demo